pkgs/applications/networking/cluster/kubeval/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, lib, fetchFromGitHub, buildGoPackage, fetchgit, makeWrapper }:
+
+let
+
+  # Cache schema as a package so network calls are not
+  # necessary at runtime, allowing use in package builds
+  schema = stdenv.mkDerivation rec {
+    name = "kubeval-schema";
+    src = fetchFromGitHub {
+      owner = "garethr";
+      repo = "kubernetes-json-schema";
+      rev = "c7672fd48e1421f0060dd54b6620baa2ab7224ba";
+      sha256 = "0picr3wvjx4qv158jy4f60pl225rm4mh0l97pf8nqi9h9x4x888p";
+    };
+
+    installPhase = ''
+      mkdir -p $out/kubernetes-json-schema/master
+      cp -R . $out/kubernetes-json-schema/master
```

Comment:
If I understand well, we can only validate against the spec of master. Would it be possible to use `$out/kubernetes-json-schema` as location? In this case, the user could choose at runtime the Kubernetes version   `--kubernetes-version`.

Reply:
No, the `--kubernetes-version` option works with the package as written.  Users can validate against many different versions.

These steps copy the entire repository, including schema for all versions, to the folder `kubernetes-json-schema/master`, because the program always adds that to its requests against `$KUBEVAL_SCHEMA_LOCATION.`  For example, if you gave the option `--kubernetes-version 1.6.6`, it would look for schema files at `file://${nix store path for schema package}/kubernetes-json-schema/master/v1.6.6-standalone/`.

The default schema location is `https://raw.githubusercontent.com/garethr`, so the `master` in this path refers to a branch in the garethr/kubernetes-json-schema repository, not in kubernetes.

Reply:
Ok. Thanks for your explanation!